pilot/pkg/xds/filters/filters.go

```diff
@@ -360,6 +361,63 @@ var (
 			}),
 		},
 	}
+	// BuildConnectOriginalDstFallbackFilter builds a filter that sets the original destination
+	// filter state. It uses the pod IP and extracts the port from the :authority header.
+	// This ensures the filter state is populated with a valid IP:port for routing.
+	BuildConnectOriginalDstFallbackFilter = func(podIP string) *hcm.HttpFilter {
+		// Template-level concatenation approach - avoid using CEL + operator
+		// Logic: If filter state key exists, use its value as-is
+		// If filter state key doesn't exist, use fallback IP with extracted port
+		// Note: Use 2 backslashes in Go raw string literal - protobuf marshaling will double them to 4 in JSON
+		// The JSON will show \\\\1 (4 backslashes), and CEL will interpret as \\1 for the regex backreference
+
+		keyCheck := fmt.Sprintf("'%s' in filter_state", OriginalDstFilterStateKey)
+
+		// Part 1: Use existing filter state value OR empty string
+		part1 := fmt.Sprintf("%%CEL(%s ? filter_state['%s'] : '')%%", keyCheck, OriginalDstFilterStateKey)
```

Comment:
Is there no way to make this one long CEL expression? Try using https://playcel.undistro.io/ to play around and see if this can be cleaned up

Reply:
I was trying out different CEL expressions with envoy's tests. One reason for this expression to be longer is that `+` operator does not work to concatenate. I even [added more tests](https://github.com/envoyproxy/envoy/pull/41607/files) in envoy to cover our case.

I will give another try